File: frontend/demo/component/formlayout/form-layout-label-styling.ts
import 'Frontend/demo/init'; // hidden-source-line
import '@vaadin/form-layout';
import '@vaadin/text-field';
import '@vaadin/email-field';
import { html, LitElement } from 'lit';
import { customElement } from 'lit/decorators.js';
import { applyTheme } from 'Frontend/demo/theme';

@customElement('form-layout-label-styling')
export class Example extends LitElement {
  protected override createRenderRoot() {
    const root = super.createRenderRoot();
    applyTheme(root);
    return root;
  }

  protected override render() {
    // tag::snippet[]
    return html`
      <vaadin-form-layout
        auto-responsive
        labels-aside
        style="
          --vaadin-form-layout-label-width: 10em;
          --vaadin-form-layout-label-spacing: 2em;
          --vaadin-form-layout-label-text-align: end;
        "
      >
        <vaadin-text-field label="First name"></vaadin-text-field>
        <vaadin-text-field label="Last name"></vaadin-text-field>
        <vaadin-email-field label="Email address"></vaadin-email-field>
      </vaadin-form-layout>
    `;
    // end::snippet[]
  }
}

File: src/main/java/com/vaadin/demo/component/formlayout/FormLayoutLabelStyling.java
package com.vaadin.demo.component.formlayout;

import com.vaadin.flow.component.formlayout.FormLayout;
// import com.vaadin.flow.component.formlayout.FormLayout.LabelTextAlign;
import com.vaadin.flow.component.html.Div;
import com.vaadin.flow.component.textfield.EmailField;
import com.vaadin.flow.component.textfield.TextField;
import com.vaadin.flow.router.Route;
import com.vaadin.demo.DemoExporter; // hidden-source-line

@Route("form-layout-label-styling")
public class FormLayoutLabelStyling extends Div {

    public FormLayoutLabelStyling() {
        // tag::snippet[]
        FormLayout formLayout = new FormLayout();
        formLayout.setAutoResponsive(true);
        formLayout.setLabelsAside(true);
        formLayout.add(new TextField("First name"), new TextField("Last name"),
                new EmailField("Email address"));

        formLayout.setLabelWidth("10em");
        formLayout.setLabelSpacing("2em");
        // formLayout.setLabelTextAlign(LabelTextAlign.END);
        // end::snippet[]

        add(formLayout);
    }

    public static class Exporter extends DemoExporter<FormLayoutLabelStyling> { // hidden-source-line
    } // hidden-source-line
}
